torch/autograd/_functions/blas.py

```diff
@@ -13,59 +13,6 @@ def _get_output(ctx, arg, inplace=False):
         return arg.new().resize_as_(arg)
 
 
-class Addmm(InplaceFunction):
```

Comment:
Where can we check the execution is through Python Function or ATen?

Reply:
It's not very good. Suppose that the operation is invoked as `x.addmm()`. Check and see if there is a `def addmm` in the `Variable` class in Python. If there is, it probably is executing through Python. If there is not, it's executing through ATen.

Another heuristic I use is grep for `Addm.apply`. If there are no such call sites, I'm pretty sure the class is dead.